### PR TITLE
FIX: Correct implementation for user preferences tracking page

### PR DESCRIPTION
**Pull request by tgxworld**
The user preferences tracking page is only present when the redesign
user navigation menu is enabled. During the first pass of
implementation, some old bugs were introduced and this commit fixes
that. Regression tests have also been added.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/controllers/preferences/tracking.js [hunk 1]
```diff
@@ -121,15 +106,19 @@ export default class extends Controller {
...
-    "model.mutedCategories"
+    "model.mutedCategories",
+    "model.regularCategories",
+    "siteSettings.mute_all_categories_by_default"
   )
   get selectedCategories() {
     return []
       .concat(
-        this.modelwatchedCategories,
+        this.model.watchedCategories,
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/app/controllers/preferences/tracking.js:116]: :bug:

File: app/assets/javascripts/discourse/tests/acceptance/user-preferences-new-menu-test.js [hunk 2]
```diff
@@ -34,24 +34,6 @@ acceptance(
       assert.ok(!exists(".nav-categories"), "Categories tab no longer exists");
     });
 
-    test("Can save category tracking preferences", async function (assert) {
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/acceptance/user-preferences-new-menu-test.js:37]: This test is redundant after the new tests have been added.

File: app/assets/javascripts/discourse/app/components/user-preferences/categories.hbs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <div class="control-group category-notifications">
   <label class="control-label">{{i18n "user.categories_settings"}}</label>
 
-  <div class="controls tracking-controls tracking-controls--watched-categories">
+  <div class="controls tracking-controls tracking-controls__watched-categories">
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/app/components/user-preferences/categories.hbs:4]: I'm following https://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/ which means this should be using `__` to represent an element. `--` is reserved for state.

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/user-preferences/categories.hbs:4]: Yes this is correct :+1:

File: app/assets/javascripts/discourse/tests/acceptance/user-preferences-tracking-test.js [hunk 4]
```diff
@@ -0,0 +1,220 @@
+import { acceptance, exists } from "discourse/tests/helpers/qunit-helpers";
+import { click, visit } from "@ember/test-helpers";
+import { test } from "qunit";
+import selectKit from "discourse/tests/helpers/select-kit-helper";
+
+acceptance("User Preferences - Tracking", function (needs) {
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/tests/acceptance/user-preferences-tracking-test.js:6]: Great tests, very nice. The middle two could be candidates for system specs as well, whereas the top and bottom ones are probably too granular.

**Review by martin-brennan — APPROVED** [review]

**Pull request merged**